frontend/src/App.js

```diff
@@ -57,26 +57,64 @@ export class App extends React.Component {
 
   handleChange = (change) => {
     switch (change.operation) {
-      case "addElement":
-        this.addElement(change.data);
+      case "addComponent":
+        this.addComponent(change.data);
         break;
       case "moveComponent":
         this.moveComponent(change.data);
         break;
       case "deleteComponent":
         this.deleteComponent(change.data);
         break;
+
+      case "addConnection":
+        this.addConnection(change.data);
+        break;
+      case "updateConnection":
+        this.updateConnection(change.data);
+        break;
+      case "deleteConnection":
+        this.deleteConnection(change.data);
+        break;
       default:
         console.error("unhandled change:", change.data);
         break;
     }
   };
 
-  addElement = (elementDetails) => {
+  addComponent = (componentDetails) => {
+    this.setState({
+      diagramData: {
+        ...this.state.diagramData,
+        components: [...this.state.diagramData.components, componentDetails],
+      },
+    });
+  };
+
+  addConnection = (connectionDetails) => {
+    console.log("connectionDetails:", connectionDetails);
```

Comment:
we should probably remove this console log

Reply:
agreed